### PR TITLE
Fix typo in dictobject comment

### PR DESCRIPTION
**Pull request by dalgarno**
Fixes a small typo in the dictobject comment. No`bpo` as it's a trivial fix.

### CODE REVIEW

**Review by brandtbucher — APPROVED** [review]

**Review by aeros — APPROVED** [review]
Thanks for the PR @dalgarno and welcome!

LGTM, we just need to wait for a core developer to review this.

Since this is involving a code comment change, I'll classify this as a documentation PR.

**Review by DinoV — APPROVED** [review]

**Comment by methane** [comment]
Oh, I just created a thread for this PR...

https://discuss.python.org/t/fixing-simple-typo-in-comment-not-document/2297

**Comment by aeros167** [comment]
@methane:
> Oh, I just created a thread for this PR...

Hmm, I can understand the concern about the messy commit logs, but I would strongly disagree that we should not accept these types of PRs. I don't think we should explicitly encourage them, but code comment typo fixing PRs provide a great introduction into the CPython workflow for new contributors. Many first time contributors start out with typo fixes, and then move on to making more impactful changes once they become more familiar with the workflow.

Also, if you want to go through the git log without seeing the typo fixes, you can simply use:

``git log --grep="typo" --invert-grep``

**Pull request merged**
